test/next_ls/dependency_test.exs

```diff
@@ -0,0 +1,141 @@
+defmodule NextLS.DependencyTest do
+  use ExUnit.Case, async: true
+
+  import GenLSP.Test
+  import NextLS.Support.Utils
+
+  @moduletag :tmp_dir
+  @moduletag root_paths: ["my_proj"]
+
+  setup %{tmp_dir: tmp_dir} do
+    File.mkdir_p!(Path.join(tmp_dir, "my_proj/lib"))
+    File.write!(Path.join(tmp_dir, "my_proj/mix.exs"), proj_mix_exs())
+
+    File.mkdir_p!(Path.join(tmp_dir, "some_dep/lib"))
+    File.write!(Path.join(tmp_dir, "some_dep/mix.exs"), dep_mix_exs())
+
+    [cwd: tmp_dir]
+  end
+
+  setup %{cwd: cwd} do
+    foo = Path.join(cwd, "my_proj/lib/foo.ex")
+
+    File.write!(foo, """
+    defmodule Foo do
+      def foo() do
+        Bar.bar()
+      end
+    end
+    """)
+
+    bar = Path.join(cwd, "some_dep/lib/bar.ex")
+
+    File.write!(bar, """
+    defmodule Bar do
+      def bar() do
+        42
+      end
+    end
+    """)
+
+    [foo: foo, bar: bar]
+  end
+
+  setup :with_lsp
+
+  test "go to dependency function definition", context do
+    %{client: client, foo: foo, bar: bar} = context
+
+    assert :ok == notify(client, %{method: "initialized", jsonrpc: "2.0", params: %{}})
+    assert_request(client, "client/registerCapability", fn _params -> nil end)
+
+    assert_is_ready(context, "my_proj")
+    assert_notification "window/logMessage", %{"message" => "[NextLS] Compiled!"}
+
+    uri = uri(foo)
+
+    request(client, %{
+      method: "textDocument/definition",
+      id: 4,
+      jsonrpc: "2.0",
+      params: %{
+        position: %{line: 2, character: 9},
+        textDocument: %{uri: uri}
+      }
+    })
+
+    uri = uri(bar)
+
+    assert_result 4, %{
+      "range" => %{
+        "start" => %{
+          "line" => 1,
+          "character" => 0
+        },
+        "end" => %{
+          "line" => 1,
+          "character" => 0
+        }
+      },
+      "uri" => ^uri
+    }
+  end
+
+  test "does not show in workspace symbols", context do
```

Comment:
still need a similar test for references

Reply:
Just to be sure, what kind of test did you have in mind here? Parent calls into dependency and the somewhere in the dependency this same function is called? Would this show up as a function reference? And what about module?

Reply:
If you do "find references" on something like Enum.map, it'll show locations in the dependencies where it's used. 

Basically when you do "find references" on anything, it should only show you references that occur inside _your_ codebase.

Reply:
Done!